crates/ruff/src/rules/pylint/rules/unexpected_special_method_signature.rs

```diff
@@ -0,0 +1,132 @@
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let verb = if self.actual_params > 1 {
+            "were"
+        } else {
+            "was"
+        };
+        if self.expected_params.end() - self.expected_params.start() > 1 {
+            format!(
+                "The special method '{}' expects between {} and {} param(s), {} {} given",
+                self.method_name,
+                self.expected_params.start(),
+                self.expected_params.end(),
+                self.actual_params,
+                verb,
+            )
+        } else {
+            format!(
+                "The special method '{}' expects {} param(s), {} {} given",
+                self.method_name,
+                self.expected_params.start(),
+                self.actual_params,
+                verb,
+            )
+        }
+    }
+}
+
+fn expected_params(name: &str, is_staticmethod: bool) -> Option<RangeInclusive<usize>> {
+    match name {
+        "__del__" | "__repr__" | "__str__" | "__bytes__" | "__hash__" | "__bool__" | "__dir__"
+        | "__len__" | "__length_hint__" | "__iter__" | "__reversed__" | "__neg__" | "__pos__"
+        | "__abs__" | "__invert__" | "__complex__" | "__int__" | "__float__" | "__index__"
+        | "__trunc__" | "__floor__" | "__ceil__" | "__enter__" | "__aenter__"
+        | "__getnewargs_ex__" | "__getnewargs__" | "__getstate__" | "__reduce__" | "__copy__"
+        | "__unicode__" | "__nonzero__" | "__await__" | "__aiter__" | "__anext__"
+        | "__fspath__" | "__subclasses__" => Some(0..=0),
+        "__format__" | "__lt__" | "__le__" | "__eq__" | "__ne__" | "__gt__" | "__ge__"
+        | "__getattr__" | "__getattribute__" | "__delattr__" | "__delete__"
+        | "__instancecheck__" | "__subclasscheck__" | "__getitem__" | "__missing__"
+        | "__delitem__" | "__contains__" | "__add__" | "__sub__" | "__mul__" | "__truediv__"
+        | "__floordiv__" | "__rfloordiv__" | "__mod__" | "__divmod__" | "__lshift__"
+        | "__rshift__" | "__and__" | "__xor__" | "__or__" | "__radd__" | "__rsub__"
+        | "__rmul__" | "__rtruediv__" | "__rmod__" | "__rdivmod__" | "__rpow__" | "__rlshift__"
+        | "__rrshift__" | "__rand__" | "__rxor__" | "__ror__" | "__iadd__" | "__isub__"
+        | "__imul__" | "__itruediv__" | "__ifloordiv__" | "__imod__" | "__ilshift__"
+        | "__irshift__" | "__iand__" | "__ixor__" | "__ior__" | "__ipow__" | "__setstate__"
+        | "__reduce_ex__" | "__deepcopy__" | "__cmp__" | "__matmul__" | "__rmatmul__"
+        | "__imatmul__" | "__div__" => Some(1..=1),
+        "__setattr__" | "__get__" | "__set__" | "__setitem__" | "__set_name__" => Some(2..=2),
+        "__exit__" | "__aexit__" => Some(3..=3),
+        "__round__" => Some(0..=1),
+        "__pow__" => Some(1..=2),
+        _ => return None,
+    }
+    .map(|r| {
+        if is_staticmethod {
+            r
+        } else {
+            RangeInclusive::new(r.start() + 1, r.end() + 1)
```

Comment:
(This could even be an associated function on `ExpectedParameters`, to increment the count.)

Reply:
`ExpectedParams::from_method()` takes an `is_staticmethod` bool and applies this internally on construction, but I got rid of the `map()` call.